### ISSUE TITLE
Minor change in model leads to huge decrease in AP

### ISSUE THREAD

**Issue by ASKanse** [facebookresearch/detectron2#1491]
## what changes I made in the code 
```
class BasicStem(CNNBlockBase):
    """
    The standard ResNet stem (layers before the first residual block).
    """

    def __init__(self, in_channels=3, out_channels=64, norm="BN"):
        """
        Args:
            norm (str or callable): norm after the first conv layer.
                See :func:`layers.get_norm` for supported format.
        """
        super().__init__(in_channels, out_channels, 4)
        self.in_channels = in_channels
        self.conv1 = Conv2d(
            in_channels,
            out_channels,
            kernel_size=7,
            stride=1,
            padding=3,
            bias=False,
            norm=get_norm(norm, out_channels),
        )
        weight_init.c2_msra_fill(self.conv1)

    def forward(self, x):
        x = self.conv1(x)
        x = F.relu_(x)
        #x = F.max_pool2d(x, kernel_size=3, stride=2, padding=1)
        return x
```
## The Issue
As seen above I changed the stride to 1, of the basic stem in resnet.py in backbone under modeling in the detectron2 folder. Also i removed the maxpooling operator. I had to make these changes as i am working on Nuclei image dataset and this down-sampling needed to be removed as it lead to immediate loss of in formation which is unfavorable.

But when I trained the model with this the following issues popped and the result that i got resolving the issues confuse me a bit.
--> first of all this lead to CUDA OOM issues in colab notebook which was resolved by only donig one of the above changes at a time.
--> the results obtained on the balloon data set showed a huge decrease in AP. It went from 73% to 0.03%. 

I expected the AP to remain in a similar range but the results counter that.      
I am curious to know what could be the reason for this, as such a minor change lead to a huge decrease in AP. A few possible reason that come to my mind are the weight initialization do not work for this stride values and a lot more epochs are needed or there is some internal constraints the model is not matching due to the change(like in RPN or ROI Heads).

Really appreciate if someone could point out the reason behind the issue or could comment on the possible reason mentioned above.

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.
